profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+    if ($cache) {
+      $taxonomy_terms = $cache->data;
+    }
+    else {
+      $query = db_select('taxonomy_term_data', 'td');
+      $query->leftJoin('taxonomy_vocabulary', 'tv', 'td.vid = tv.vid');
+      $query->fields('td');
+      $query->fields('tv', array('machine_name'));
+      $query->condition('tv.machine_name', $vocabularies);
+      $taxonomy_terms = NewsroomHelper::convertPdoToArray($query->execute());
+      cache_set($cache_key, $taxonomy_terms, NEXTEUROPA_NEWSROOM_CACHE_TABLE, time() + 3600);
+    }
+    return $taxonomy_terms;
+  }
+
+  /**
+   * Try to get taxonomy URL part (taxonomy term name).
+   *
+   * @param string $request_name
+   *   URL request part.
+   * @param array $vocabularies
+   *   List of vocabularies.
+   */
+  public static function getTermFromRequest($request_name, $vocabularies) {
+    $cache_key = 'newsroom:term_from_request:' . md5($request_name);
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $taxonomy_term = $cache->data;
+    }
+    else {
+      $taxonomy_term = NULL;
+      if (is_int($request_name)) {
+        $taxonomy_term = taxonomy_term_load($request_name);
+      }
+
+      if (!$taxonomy_term && count($vocabularies) > 0) {
+        $taxonomy_terms = NewsroomHelper::getTermsFromVocabularies($vocabularies);
+        // Compare taxonomy terms modified by pathauto with the current value
+        // from the URL.
+        if ($taxonomy_terms) {
+          foreach ($taxonomy_terms as $term_item) {
+            // Replace the name with translated value.
+            $fullterm = taxonomy_term_load($term_item['tid']);
+            $term_item['name'] = NewsroomHelper::getTermTranslatedName($fullterm);
+            if (pathauto_cleanstring($term_item['name']) == $request_name) {
+              $taxonomy_term = $fullterm;
+              break;
+            }
+          }
+        }
+      }
+      cache_set($cache_key, $taxonomy_term, NEXTEUROPA_NEWSROOM_CACHE_TABLE, time() + 3600);
+    }
+    return $taxonomy_term;
+  }
+
+  /**
+   * Gets translated taxonomy term name.
+   *
+   * @param object $taxonomy_term
+   *   Taxonomy term object.
+   *
+   * @return string
+   *   The name of taxonomy term.
+   */
+  public static function getTermTranslatedName($taxonomy_term) {
+    global $language;
+    $curr_lang = $language->language;
+    $translation = field_get_items('taxonomy_term', $taxonomy_term, 'name_field', $curr_lang);
+    return $translation ? $translation[0]['safe_value'] : $taxonomy_term->name;
+  }
+
+  /**
+   * Defines whether the item is new or not.
+   *
+   * @param int $created
+   *   Timestamp when item was created.
+   * @param int $nid
+   *   Node id.
+   *
+   * @return bool
+   *   It is a new item or not.
+   */
+  public static function isNewItem($created, $nid) {
+    return user_is_logged_in() && $created > time() - 14 * 86400 && !node_last_viewed($nid);
```

Comment:
are you sure this meaning of something is new won't change?! you should not hardcode a value like you do it here, but put it to a variable what can be edited from an admin form.